src/main/java/org/commonwl/view/git/GitService.java

```diff
@@ -84,34 +86,45 @@ public Git getRepository(GitDetails gitDetails, boolean reuseDir)
             } else {
                 // Create a folder and clone repository into it
                 Files.createDirectory(repoDir);
-                repo = Git.cloneRepository()
-                        .setCloneSubmodules(cloneSubmodules)
-                        .setURI(gitDetails.getRepoUrl())
-                        .setDirectory(repoDir.toFile())
-                        .setCloneAllBranches(true)
-                        .call();
+                repo = cloneRepo(gitDetails.getRepoUrl(), repoDir.toFile());
             }
         } else {
             // Another thread is already using the existing folder
             // Must create another temporary one
-            repo = Git.cloneRepository()
-                    .setCloneSubmodules(cloneSubmodules)
-                    .setURI(gitDetails.getRepoUrl())
-                    .setDirectory(createTempDir())
-                    .setCloneAllBranches(true)
-                    .call();
+            repo = cloneRepo(gitDetails.getRepoUrl(), createTempDir());
         }
 
         // Checkout the specific branch or commit ID
         if (repo != null) {
             // Create a new local branch if it does not exist and not a commit ID
             String branchOrCommitId = gitDetails.getBranch();
-            if (!ObjectId.isId(branchOrCommitId)) {
+            final boolean isId = ObjectId.isId(branchOrCommitId);
+            if (!isId) {
                 branchOrCommitId = "refs/remotes/origin/" + branchOrCommitId;
             }
-            repo.checkout()
-                    .setName(branchOrCommitId)
-                    .call();
+            try {
```

Comment:
Agree on the logic of trying branch before tag, that's the logic used at GitHub as well - see https://github.com/stain/branchandtagtest/blob/test/README.md where I tried to make a branch and a tag both called `test` (but the branch further ahead)